x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapperTests.java

```diff
@@ -439,7 +439,7 @@ private SemanticTextIndexOptions getDefaultSparseVectorIndexOptionsForMapper(Map
 
     public void testInvalidTaskTypes() {
         for (var taskType : TaskType.values()) {
-            if (taskType == TaskType.TEXT_EMBEDDING || taskType == TaskType.SPARSE_EMBEDDING) {
+            if (taskType == TaskType.TEXT_EMBEDDING || taskType == TaskType.SPARSE_EMBEDDING || taskType == TaskType.EMBEDDING) {
```

Comment:
this test made me think initially that we made some changes to `SemanticTextFieldMapper` to support this task type. this is obviously not the case.

I wonder if instead of skipping the `TaskType.EMBEDDING` we can assert the real exception, or add at least add a comment on why we are skipping EMBEDDING here.